unleash/src/main/java/com/silvercar/unleash/UnleashContext.kt

```diff
@@ -1,7 +1,10 @@
 package com.silvercar.unleash
 
+import com.silvercar.unleash.UnleashContext.Builder
 import com.silvercar.unleash.util.UnleashConfig
 
+fun unleashContext(func: Builder.() -> Unit) = Builder().apply(func)
```

Comment:
In Kotlin functions can be declared at top level in a file. This doesn't exist in Java and these functions can't be used in Java files so it isn't exactly the same as static.

Reply:
Hmmm, now I’m curious as to why you declared it this way. What benefit do we gain by doing so?

Reply:
Here is the [presentation](https://academy.realm.io/posts/kau-jake-wharton-testing-robots/) where I learned this pattern but this is a way to create a factory function. So instead of having to write `UnleashContext.create()` you can now just write `unleashContext { }`.

The equivalent to Java's static is Kotlin's `companion object`. You can even annotate these with `@JvmStatic` to make them behave the same in Java files.

Reply:
The main benefit I saw in this pattern was being able to write more concise code.  I might be able to make this even more concise by automatically calling `.build()` at the end but I haven't tried that yet.